### PR TITLE
refactor: migrates tests to jest

### PR DESCRIPTION
**Pull request by schne324**

### CODE REVIEW

**Comment by schne324** [comment]
I was getting kind of annoyed with tape so I decided to migrate to jest. I think jest is something more of the team is familiar with and comes with built-in reporting/coverage/watcher/other fancy things.

**Comment by schne324** [comment]
*NOTE*: I did not add any new code here, just new dependencies and converted existing tests

File: .eslintrc.json [hunk 1]
```diff
@@ -5,6 +5,15 @@
     "node": true,
     "es6": true
   },
+  "globals": {
+    "jest": true,
```
Comment (stephenmathieson) [inline .eslintrc.json]: These globals should only be "allowed" in test files. You can do this with "overrides":

```json
"overrides": [
  {
    "files": "*.test.js",
    "globals": {
      "jest": true,
      "test": true,
      "expect": true,
      ...
    }
  }
]
```

Comment (stephenmathieson) [reply .eslintrc.json]: It looks like your tests are in `__tests__`, so the `files` glob will need to reflect that.

**Pull request merged**